api/src/main/java/com/datatorrent/api/Context.java

```diff
@@ -301,6 +301,12 @@
      */
     int getId();
 
+    /**
+     * Return the number of windows before the next checkpoint including the current window.
+     * @return Number of windows from checkpoint, 1 if the checkpoint will be after the current window
+     */
+    int getWindowsFromCheckpoint();
```

Comment:
@PramodSSImmaneni This is a backward incompatible change. Did you consider to extend this interface, add say DelayContext and then add the method in there?

Also semver plugin should be catching this right?